src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -1875,8 +1875,13 @@ private FileSystemObject GetFileSystemObjectForDirectory(string path)
         private string GetModelName(string org, string app)
         {
             PlatformStorageModels.Application application = GetApplication(org, app);
-
             string dataTypeId = string.Empty;
+
+            if (application == null)
```

Comment:
Is this a state we shouldn't get into? Did you figure out why we end up with application being null?

Reply:
Yeah, I added som info under cause on the issue. This request is made every time a repository is cloned to a user's personal area in designer. Will create an issue for not making the request when it is not nessesary, but for now this will solve the null exceptions in our logs.

Reply:
Ok, I see. I think option B is the way to go, and I've just added code that allows us to check for the type of repo so we can use that in these kind of flows to avoid doing "app" stuff in datamodels repo and vice versa.

Reply:
Timing wise, we probably have other issues that take presedence. So, merging this fix would make sense at this point to avoid exceptions. Our logic should be able to handle null values in a reasonable way regardless of which changes are made in other parts of the solution.